src/node.ts

```diff
@@ -55,11 +58,13 @@ class Node extends VEventTarget {
   linkInput(name: string, outputSocket: OutputSocket): void {
     const input = this.getInput(name);
     input.linkSocket(outputSocket);
+    this.dispatchEvent(new VEvent('linkInput'));
   }
 
   unlinkInput(name: string): void {
     const input = this.getInput(name);
     input.clearLink();
+    this.dispatchEvent(new VEvent('unlinkInput'));
```

Comment:
Without extra info the event handler will not have idea which input was unlinked. Perhaps the input name should be added as extra data.